### PR TITLE
Add create as qa default category setting

### PR DESCRIPTION
**Pull request by angusmcleod**
@techAPJ This is a draft for "[Add setting QuestionAnswer at the category level](https://meta.discourse.org/t/add-setting-questionanswer-at-the-category-level/235040)". Just checking we're on the same page.

### CODE REVIEW

**Comment by techAPJ** [comment]
I read through the code and tested it locally, and the implementation looks good to me. 👍🏼

**Comment by angusmcleod** [comment]
@techAPJ Ok! Making this a proper PR.

File: plugin.rb [hunk 1]
```diff
@@ -191,4 +191,13 @@
 
     false
   end
+
+  register_category_custom_field_type('create_as_qa_default', :boolean)
```
Comment (techAPJ) [inline plugin.rb:195]: I think we should use constant here similar to how we did in [chat plugin](https://github.com/discourse/discourse-chat/blob/9c4271b846aa03d7bce6e553dc068043e08c9e9c/plugin.rb#L203) and then reuse that constant.

File: plugin.rb [hunk 2]
```diff
@@ -191,4 +191,13 @@
...
+  if Site.respond_to? :preloaded_category_custom_fields
+    Site.preloaded_category_custom_fields << 'create_as_qa_default'
+  end
+  add_to_class(:category, :create_as_qa_default) do
+    ActiveModel::Type::Boolean.new.cast(self.custom_fields['create_as_qa_default'])
+  end
+  add_to_serializer(:basic_category, :create_as_qa_default) { object.create_as_qa_default }
```
Comment (techAPJ) [inline plugin.rb:202]: Can we add a test to verify that the correct value is present in the serializer?

**Review by techAPJ — CHANGES_REQUESTED** [review]
Looks good, added some minor feedback.

File: plugin.rb (hunk 1 above)
Comment (angusmcleod) [reply plugin.rb]: Added https://github.com/discourse/discourse-question-answer/pull/96/commits/dc72d4456422ba58a4472b059fbb902efc0264b7

File: plugin.rb (hunk 2 above)
Comment (angusmcleod) [reply plugin.rb:202]: Also added! https://github.com/discourse/discourse-question-answer/pull/96/commits/f13168aef0a29ad63bc2b472ea6ce787371787c6

**Comment by angusmcleod** [comment]
@techAPJ Thanks, done :)

**Review by techAPJ — APPROVED** [review]
Thank you! ❤️

**Pull request merged**